src/nvim/ex_cmds2.c

```diff
@@ -1942,10 +1950,12 @@ void ex_listdo(exarg_T *eap)
         ex_cc(eap);
 
         buf = curbuf;
-        i = eap->line1 - 1;
+        assert(eap->line1 <= INT_MAX);
+        i = (int)eap->line1 - 1;
         if (eap->addr_count <= 0) {
           // Default to all quickfix/location list entries.
-          eap->line2 = qf_size;
+          assert(qf_size <= LONG_MAX);
```

Comment:
`linenr_T` is a type alias to `long`, and `qf_size` is `unsigned long` so I added this assertion. However it is brittle, if `linenr_T` changes, this will have to be updated. Is there a better way to do this?